### PR TITLE
feat: support multi private key sign transaction

### PR DESCRIPTION
**Pull request by duanyytop**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,90 @@ Block block = ckbService.getBlock(blockHash).send().getBlock();
...
+                "a202386cb9e46cecff9bc14b748b714c713075dd964c2507c8a8900540164959",
+                "ckt1qyqtnz38fht9nvmrfdeunrhdtp29n0gagkps4duhek"),
+            new Account(
+                "89b773ec5cf97b8fd2cf280ab1e37cd658dc28d84bac8f8dda4a8646cc08d266",
+                "ckt1qyqxvnycu7tdtyuejn3mmcnl4y09muxz8c3s2ewjd4"));
+  }
+
+  public static void main(String[] args) throws Exception {
+    String minerPrivateKey = "e79f3207ea4980b7fed79956d5934249ceac4751a4fae01a0f7c4a96884bc4e3";
+    List<Receiver> funds =
```
Comment (shaojunda) [inline README.md]: why the variable name is `funds` but it type is Receiver?

File: README.md [hunk 2]
```diff
@@ -54,6 +54,90 @@ Block block = ckbService.getBlock(blockHash).send().getBlock();
...
+                "ckt1qyq2pmuxkr0xwx8kp3ya2juryrygf27dregs44skek",
+                new BigInteger("600").multiply(UnitCKB)));
+
+    System.out.println(
+        "Before transfer, first receiver's balance: "
+            + getBalance(receivers.get(0).address).divide(UnitCKB).toString(10)
+            + " CKB");
+
+    // fund accounts send capacity to three receiver accounts with 4000, 15000 and 2000 CKB
+    sendCapacityWithMultiPrivateKey(fundSenders, receivers);
```
Comment (shaojunda) [inline README.md]: senders and receivers, or fundSenders and fundReceivers?

File: README.md (hunk 1 above)
Comment (duanyytop) [reply README.md]: funds mean that they have more CKB. And they will send capcity to other accounts, so it is just a account name.

File: README.md (hunk 2 above)
Comment (duanyytop) [reply README.md]: `fundSenders` are just keep accordance to above funds and `receivers` are just general receive accounts.

File: core/src/main/java/org/nervos/ckb/transaction/TxGenerator.java [hunk 3]
```diff
@@ -19,40 +21,76 @@
...
+    for (Sender sender : senders) {
+      String lockHash =
+          Utils.generateLockScriptWithPrivateKey(sender.privateKey, systemSecpCell.cellHash)
+              .computeHash();
+      CollectedCells collectedCells =
+          new CellGatherer(ckbService).getCellInputs(lockHash, sender.capacity);
+      if (collectedCells.capacity.compareTo(sender.capacity) < 0) {
+        throw new IOException("No enough Capacities with sender private key: " + sender.privateKey);
+      }
+      collectCapacity = collectCapacity.add(collectedCells.capacity);
```
Comment (shaojunda) [inline core/src/main/java/org/nervos/ckb/transaction/TxGenerator.java]: collectedCapacity may be better?

Comment (duanyytop) [reply core/src/main/java/org/nervos/ckb/transaction/TxGenerator.java]: Updated. https://github.com/nervosnetwork/ckb-sdk-java/pull/170/commits/7ad9f29f89b60ae4cb3cd7d527edafcf66c88c8e

File: README.md (hunk 1 above)
Comment (shaojunda) [reply README.md]: ok

File: core/src/main/java/org/nervos/ckb/transaction/TxGenerator.java [hunk 4]
```diff
@@ -61,18 +99,20 @@ public Transaction generateTx(List<Receiver> receivers) throws IOException {
           new CellOutput(
               receiver.capacity.toString(),
               new Script(
-                  systemScriptCell.cellHash, Collections.singletonList(blake2b), Script.TYPE)));
+                  systemSecpCell.cellHash, Collections.singletonList(blake2b), Script.TYPE)));
     }
 
-    if (cellInputs.capacity.compareTo(needCapacities) > 0) {
+    // set first sender's public key blake160 to change cell output
+    if (collectedCapacity.compareTo(needCapacities) > 0) {
```
Comment (shaojunda) [inline core/src/main/java/org/nervos/ckb/transaction/TxGenerator.java]: not sure if this strategy is suitable here.

File: README.md (hunk 2 above)
Comment (duanyytop) [reply README.md]: Updated. https://github.com/nervosnetwork/ckb-sdk-java/pull/170/commits/fd1a27e6e9b62241b184467523cb9384b5d5c0f6

File: README.md (hunk 1 above)
Comment (duanyytop) [reply README.md]: Updated. https://github.com/nervosnetwork/ckb-sdk-java/pull/170/commits/fd1a27e6e9b62241b184467523cb9384b5d5c0f6

**Comment by ashchan** [comment]
Marking this as **HOLD** as I would like to see a major simplifying.

**Comment by duanyytop** [comment]
@ashchan @shaojunda  Refactored.

**Pull request merged**